web/src/enterprise/site-admin/dotcom/productSubscriptions/SiteAdminCreateProductSubscriptionPage.tsx

```diff
@@ -1,244 +1,175 @@
-import { gql } from '../../../../../../shared/src/graphql/graphql'
+import * as H from 'history'
+import AddIcon from 'mdi-react/AddIcon'
+import React, { useCallback } from 'react'
+import { Redirect, RouteComponentProps } from 'react-router'
+import { Link } from 'react-router-dom'
+import { concat, Observable } from 'rxjs'
+import { catchError, concatMap, map, tap } from 'rxjs/operators'
+import { dataOrThrowErrors, gql } from '../../../../../../shared/src/graphql/graphql'
 import * as GQL from '../../../../../../shared/src/graphql/schema'
-import { asError, createAggregateError, ErrorLike, isErrorLike } from '../../../../../../shared/src/util/errors'
-import { pluralize } from '../../../../../../shared/src/util/strings'
+import { asError } from '../../../../../../shared/src/util/errors'
 import { mutateGraphQL, queryGraphQL } from '../../../../backend/graphql'
+import { FilteredConnection } from '../../../../components/FilteredConnection'
 import { Form } from '../../../../components/Form'
 import { PageTitle } from '../../../../components/PageTitle'
 import { eventLogger } from '../../../../tracking/eventLogger'
-import { AccountEmailAddresses } from '../../../dotcom/productSubscriptions/AccountEmailAddresses'
-import { ErrorAlert } from '../../../../components/alerts'
+import { useEventObservable } from '../../../../util/useObservable'
 
-interface Props extends RouteComponentProps<{}> {
-    authenticatedUser: GQL.IUser
-}
+interface UserCreateSubscriptionNodeProps {
+    /**
+     * The user to display in this list item.
+     */
+    node: GQL.IUser
 
-/** A customer account. */
-interface Account extends Pick<GQL.IUser, 'id' | 'username' | 'displayName'> {
-    emails: Pick<GQL.IUserEmail, 'email' | 'verified'>[]
+    /**
+     * Browser history, used to redirect the user to the new subscription after one is successfully created.
+     */
+    history: H.History
 }
 
-const LOADING: 'loading' = 'loading'
+const createProductSubscription = (
+    args: GQL.ICreateProductSubscriptionOnDotcomMutationArguments
+): Observable<Pick<GQL.IProductSubscription, 'urlForSiteAdmin'>> =>
+    mutateGraphQL(
+        gql`
+            mutation CreateProductSubscription($accountID: ID!) {
+                dotcom {
+                    createProductSubscription(accountID: $accountID) {
+                        urlForSiteAdmin
+                    }
+                }
+            }
+        `,
+        args
+    ).pipe(
+        map(dataOrThrowErrors),
+        map(data => data.dotcom.createProductSubscription)
+    )
+
+const UserCreateSubscriptionNode: React.FunctionComponent<UserCreateSubscriptionNodeProps> = (
+    props: UserCreateSubscriptionNodeProps
+) => {
+    const [onSubmit, createdSubscription] = useEventObservable(
+        useCallback(
+            (
+                submits: Observable<React.FormEvent<HTMLFormElement>>
+            ): Observable<Pick<GQL.IProductSubscription, 'urlForSiteAdmin'> | 'saving' | Error> =>
```

Comment:
We typically use `ErrorLike` along with the `isErrorLike` typeguard (rather than `x instanceof Error`).

```suggestion
            ): Observable<Pick<GQL.IProductSubscription, 'urlForSiteAdmin'> | 'saving' | ErrorLike> =>
```